test/e2e/magic_catalog_test.go

```diff
@@ -27,15 +28,43 @@ var _ = Describe("MagicCatalog", func() {
 		// create dependencies
 		const catalogName = "test"
 		namespace := generatedNamespace.GetName()
-		kubeClient := ctx.Ctx().Client()
-		provider, err := NewFileBasedFiledBasedCatalogProvider(filepath.Join(testdataDir, "fbc_catalog.json"))
+		provider, err := NewFileBasedFiledBasedCatalogProvider(filepath.Join(testdataDir, "magiccatalog/fbc_catalog.json"))
 		Expect(err).To(BeNil())
 
 		// create and deploy and undeploy the magic catalog
-		magicCatalog := NewMagicCatalog(kubeClient, namespace, catalogName, provider)
+		magicCatalog := NewMagicCatalog(c, namespace, catalogName, provider)
 
 		// deployment blocks until the catalog source has reached a READY status
-		Expect(magicCatalog.DeployCatalog(context.TODO())).To(BeNil())
-		Expect(magicCatalog.UndeployCatalog(context.TODO())).To(BeNil())
+		Expect(magicCatalog.DeployCatalog(context.Background())).To(BeNil())
+		Expect(magicCatalog.UndeployCatalog(context.Background())).To(BeNil())
+	})
+	When("an existing magic catalog exists", func() {
+		var (
+			mc          MagicCatalog
+			catalogName string
+		)
+		BeforeEach(func() {
+			provider, err := NewFileBasedFiledBasedCatalogProvider(filepath.Join(testdataDir, "magiccatalog/fbc_initial.yaml"))
+			Expect(err).To(BeNil())
+
+			catalogName = genName("mc-e2e-catalog-")
+
+			mc = NewMagicCatalog(c, generatedNamespace.GetName(), catalogName, provider)
+			Expect(mc.DeployCatalog(context.Background())).To(BeNil())
+		})
+		AfterEach(func() {
+			Expect(mc.UndeployCatalog(context.Background())).To(BeNil())
+		})
+
+		It("should succeed when the magic catalog is updated", func() {
+			updatedProvider, err := NewFileBasedFiledBasedCatalogProvider(filepath.Join(testdataDir, "magiccatalog/fbc_updated.yaml"))
+			Expect(err).To(BeNil())
+
+			updatedMC := NewMagicCatalog(c, generatedNamespace.GetName(), catalogName, updatedProvider)
+			Expect(updatedMC.UpdateCatalog(context.Background(), updatedProvider)).To(BeNil())
```

Comment:
is it worth checking that the contents of the catalog got updated somehow:?

Reply:
Yep - I had briefly thought about this yesterday when poking at the update logic. I was tempted to verify the grpc contents directly but that didn't feel very representative of an e2e test. I think validating that the configmap contents have changed might be sufficient? Thoughts?

Reply:
Yeah - I'm good with that. Just a sanity check. You could also check against a check sum or something.

Reply:
What did you have in mind for a check sum implementation? That seems like an interesting angle, but I ended up settling with comparing the configmap catalog.json directly with our expectation of those contents.